django/db/backends/postgresql/base.py

```diff
@@ -212,7 +260,17 @@ def get_connection_params(self):
 
     @async_unsafe
     def get_new_connection(self, conn_params):
```

Comment:
As Simon suggested during the DjangoCon US 2022 sprints since the psycopg version is known at class definition time , maybe we should define the psycopg specific implementation at class definition time instead of at runtime?

Reply:
I fear I am not exactly sure what you mean.

Reply:
@charettes wrote in this comment that he's working on it
https://github.com/django/django/pull/15687#issuecomment-1289523684